lib/evaluate.js

```diff
@@ -179,6 +183,11 @@ export const evaluate = async ({
   }
   const setScoresDuration = Date.now() - start
 
+  ;(async () => {
+    await tx.wait()
+    stuckTransactionsCanceller.removeConfirmed(tx)
```

Comment:
IIUC (see my comment above), you are going to call `removeConfirmed` only the last transaction. That does not look correct to me. IMO, we need to remove all confirmed transactions created in this round.